### PR TITLE
#926 - Converting all inputs which are coming as secret to lowercase

### LINKED ISSUE #926: Refactor Secret Name
make Secret names caseinsensitive ...
- [ ] add migration to store them all lowercase in backend
- [ ] convert all inputs to lowercase

### PR DESCRIPTION
**Pull request by avinilcode**
Closes #926

### CODE REVIEW

**Comment by avinilcode** [comment]
ok, will try to do it.

**Comment by avinilcode** [comment]
please review now, i will add for migrating it, if there are any questions i will ask

File: server/api/repo_secret.go [hunk 1]
```diff
@@ -95,6 +96,7 @@ func PatchSecret(c *gin.Context) {
 		secret.Images = in.Images
 	}
 
+	secret.Name = strings.ToLower(in.Name)
```
[the remark this answers is not in the document]
Comment (avinilcode) [reply server/api/repo_secret.go]: done

**Comment by avinilcode** [comment]
> Yes, that's what I mean. Just the migration is missing :) You can take a look at https://github.com/woodpecker-ci/woodpecker/tree/master/server/store/datastore/migration

How can i go about it, i have to first fetch all the records in DB, i tried but not sure.

**Comment by avinilcode** [comment]
> Take a look at the other migrations (copy one of the simpler ones like https://github.com/woodpecker-ci/woodpecker/blob/master/server/store/datastore/migration/006_log_data_type.go) and run an SQL statement like
> 
> ```sql
> UPDATE secrets SET secret_name = LOWER(secret_name)
> ```
> 
> which sets all values to lowercase. https://stackoverflow.com/questions/6160873/update-all-values-of-a-column-to-lowercase

**Comment by avinilcode** [comment]
@qwerty287 Please review it now

**Comment by avinilcode** [comment]
> Did you close this accidentally?

yes

File: server/store/datastore/migration/008_secrets_update_secret_name.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	Name   string `json:"name" xorm:"NOT NULL UNIQUE(s) INDEX 'secret_name'"`
+}
+
+// TableName return database table name for xorm
+func (SecretV008) TableName() string {
+	return "secrets"
+}
+
+var updateTableSecretsUpdateSecretName = task{
+	name: "update-table-secrets-secret-name",
```
[the remark this answers is not in the document]
Comment (avinilcode) [reply server/store/datastore/migration/008_secrets_update_secret_name.go]: done

**Comment by avinilcode** [comment]
Thank you so much for quick reviewing and replies, it is my first PR in open source, means a lot

**Comment by avinilcode** [comment]
can you please merge it.

**Comment by avinilcode** [comment]
> Sure.CI passed, so let's go. Thank you again tada You're welcome to take a look at more issues if you want :)

Sure, thank you very much!

**Comment by 6543** [comment]
thansk for the pul @avinilcode thanks for the reviews @qwerty287 :)

**Pull request closed without merging**